crates/dbsp/src/operator/communication/shard.rs

```diff
@@ -88,7 +90,7 @@ where
     #[track_caller]
     pub fn shard(&self) -> Stream<C, IB>
     where
-        IB: Batch + Send,
+        IB: Batch + Send + Encode + Decode,
```

Comment:
It might be a good idea to add the `Encode + Decode` bound to `trait BatchReader` or `trait Batch`, since they are required by most operators.

Reply:
Done.